source/server-discovery-and-monitoring/server-discovery-and-monitoring-tests.rst

```diff
@@ -178,3 +178,35 @@ Attempt a write to a collection.
 
 Outcome: Verify that the write succeeded or failed depending on existing
 driver behavior with respect to the starting topology.
+
+Connection Pool Management
```

Comment:
The Rust driver doesn't have SDAM monitoring yet, so I could only implement a portion of this test. Let me know how the full implementation goes in Python.

Reply:
It works with the changes I suggested below. We should also mention this test requires failCommand appName support which is now >=4.2.9 (see SERVER-47195).

Reply:
Done

Reply:
Could this test be written as an SDAM integration spec test?

Reply:
The key assertion that "ConnectionPoolReadyEvent occurs after the first ServerHeartbeatSucceededEvent" is not possible in the SDAM test format. So I would say no.